--- rpmlint-1.7/rpmlint.threshold~	2015-07-08 11:22:46.000000000 +0200
+++ rpmlint-1.7/rpmlint	2016-08-19 10:17:40.465997226 +0200
@@ -94,6 +94,7 @@ def main():
 
     packages_checked = 0
     specfiles_checked = 0
+    threshold_exceeded = False
 
     try:
         # Loop over all file names given in arguments
@@ -194,17 +195,24 @@ def main():
                 continue
 
         if printAllReasons():
-            Pkg.warn('(none): E: badness %d exceeds threshold %d, aborting.' %
-                     (badnessScore(), badnessThreshold()))
-            sys.exit(66)
+            if below_threshold_returns_zero:
+                threshold_exceeded = True
+            else:
+                Pkg.warn('(none): E: badness %d exceeds threshold %d, aborting.' %
+                         (badnessScore(), badnessThreshold()))
+                sys.exit(66)
 
     finally:
         print("%d packages and %d specfiles checked; %d errors, %d warnings."
               % (packages_checked, specfiles_checked,
                  printed_messages["E"], printed_messages["W"]))
 
-    if printed_messages["E"] > 0:
+    if not below_threshold_returns_zero and printed_messages["E"] > 0:
         sys.exit(64)
+    if below_threshold_returns_zero and threshold_exceeded:
+        Pkg.warn('(none): E: badness %d exceeds threshold %d.' %
+                 (badnessScore(), badnessThreshold()))
+        sys.exit(66)
     sys.exit(0)
 
 
@@ -244,7 +252,7 @@ argv0 = os.path.basename(sys.argv[0])
 # parse options
 try:
     (opt, args) = getopt.getopt(sys.argv[1:],
-                              'iI:c:C:hVvanE:f:o:',
+                              'iI:c:C:hVvanE:f:o:T',
                               ['info',
                                'explain=',
                                'check=',
@@ -258,6 +266,7 @@ try:
                                'file=',
                                'option=',
                                'rawout=',
+                               'zerothreshold',
                                ])
 except getopt.GetoptError:
     Pkg.warn("%s: %s" % (argv0, sys.exc_info()[1]))
@@ -274,6 +283,7 @@ if not os.path.exists(os.path.expanduser
     # deprecated backwards compatibility with < 0.88
     conf_file = '~/.rpmlintrc'
 info_error = set()
+below_threshold_returns_zero = False
 
 # load global config files
 configs = glob.glob('/etc/rpmlint/*config')
@@ -329,6 +339,9 @@ for o in opt:
         Config.setOption('ExtractDir', extract_dir)
     elif o[0] in ('-n', '--noexception'):
         Config.no_exception = True
+    # terrible longname, but whatever..
+    elif o[0] in ('-T', 'zerothreshold'):
+        below_threshold_returns_zero = True
     elif o[0] in ('-a', '--all'):
         if '*' not in args:
             args.append('*')
