diff -up rpmlint-rpmlint-1.11/rpmlint.omv~ rpmlint-rpmlint-1.11/rpmlint
--- rpmlint-rpmlint-1.11/rpmlint.omv~	2019-03-14 19:25:50.884537337 +0100
+++ rpmlint-rpmlint-1.11/rpmlint	2019-03-14 19:27:33.505971353 +0100
@@ -89,7 +89,13 @@ def loadCheck(name):
 #############################################################################
 def main():
 
-    locale.setlocale(locale.LC_COLLATE, '')
+    try:
+        locale.setlocale(locale.LC_COLLATE, '')
+    except locale.Error:
+        # This happens e.g. while bootstrapping a system
+        # with glibc built --without i18ndata
+        # Let's not fail the build.
+        pass
 
     # Add check dirs to the front of load path
     sys.path[0:0] = Config.checkDirs()
