--- rpmlint-1.7/config.info~	2016-08-19 11:51:45.278071086 +0200
+++ rpmlint-1.7/config	2016-08-19 11:56:08.539097697 +0200
@@ -212,6 +212,12 @@ from Config import *
 # Type: string, default: 'mac'
 #SetOption("ManWarningCategory", 'mac')
 
+# Whether install-info of texinfo documentation is handled by triggers.
+# If handled by triggers, report if install-info is used in scriptlets and
+# also any dependencies on it.
+# Type: boolean, default: False
+#setOption("InstallInfoTriggers", False)
+
 # Command and arguments to validate appdata files.
 # Type: tuple of strings, default: see DEFAULT_APPDATA_CHECKER in AppDataCheck
 #setOption("AppDataChecker", ('appstream-util', 'validate-relax'))
--- rpmlint-1.7/FilesCheck.py.info~	2015-07-08 15:57:33.000000000 +0200
+++ rpmlint-1.7/FilesCheck.py	2016-08-19 12:00:05.871113687 +0200
@@ -228,6 +228,7 @@ skipdocs_regex = re.compile(Config.getOp
 meta_package_regex = re.compile(Config.getOption('MetaPackageRegexp', '^(bundle|task)-'))
 filesys_packages = ['filesystem']  # TODO: make configurable?
 quotes_regex = re.compile('[\'"]+')
+install_info_triggers = Config.getOption('InstallInfoTriggers', False)
 start_certificate_regex = re.compile('^-----BEGIN CERTIFICATE-----$')
 start_private_key_regex = re.compile('^----BEGIN PRIVATE KEY-----$')
 
@@ -621,21 +622,32 @@ class FilesCheck(AbstractCheck.AbstractC
 
                 # check install-info call in %post and %postun
                 if f.startswith('/usr/share/info/'):
-                    if not postin:
-                        printError(pkg,
-                                   'info-files-without-install-info-postin', f)
-                    elif not install_info_regex.search(postin):
-                        printError(pkg, 'postin-without-install-info', f)
+                    if not install_info_triggers:
+                        if not postin:
+                            printError(pkg,
+                                       'info-files-without-install-info-postin', f)
+                        elif not install_info_regex.search(postin):
+                            printError(pkg, 'postin-without-install-info', f)
+                    else:
+                        if postin and install_info_regex.search(postin):
+                            printError(pkg,
+                                       'info-files-with-install-info-postin', f)
 
                     preun = pkg[rpm.RPMTAG_PREUN] or \
                             pkg.scriptprog(rpm.RPMTAG_PREUNPROG)
-                    if not postun and not preun:
-                        printError(pkg,
-                                   'info-files-without-install-info-postun', f)
-                    elif (not postun or
-                          not install_info_regex.search(postun)) and \
-                          (not preun or not install_info_regex.search(preun)):
-                        printError(pkg, 'postin-without-install-info', f)
+                    if not install_info_triggers:
+                        if not postun and not preun:
+                            printError(pkg,
+                                       'info-files-without-install-info-postun', f)
+                        elif (not postun or
+                              not install_info_regex.search(postun)) and \
+                              (not preun or not install_info_regex.search(preun)):
+                            printError(pkg, 'postin-without-install-info', f)
+                    else:
+                        if (postun and install_info_regex.search(postun)) or \
+                                (preun and install_info_regex.search(preun)):
+                            printError(pkg,
+                                       'info-files-with-install-info-postun', f)
 
                 # check perl temp file
                 if perl_temp_file_regex.search(f):
@@ -1030,6 +1042,11 @@ a call to ldconfig.''',
 'postun-without-ldconfig',
 '''This package contains a library and its %postun doesn't call ldconfig.''',
 
+'info-files-with-install-info-postin',
+'''While this package contains info files and provides a %post scriptlet
+containing a call to install-info, this is no longer required due to file
+trigger replacement.''',
+
 'info-files-without-install-info-postin',
 '''This package contains info files and provides no %post scriptlet containing
 a call to install-info.''',
@@ -1037,6 +1054,11 @@ a call to install-info.''',
 'postin-without-install-info',
 'This package contains info files and its %post doesn\'t call install-info.',
 
+'info-files-with-install-info-postun',
+'''While this package contains info files and provides a %postun scriptlet
+containing a call to install-info, this is no longer required due to file
+trigger replacement.''',
+
 'info-files-without-install-info-postun',
 '''This package contains info files and provides no %postun scriptlet
 containing a call to install-info.''',
--- rpmlint-1.7/TagsCheck.py.info~	2016-08-19 11:51:45.271071109 +0200
+++ rpmlint-1.7/TagsCheck.py	2016-08-19 11:51:45.278071086 +0200
@@ -25,6 +25,8 @@ import FilesCheck
 import Pkg
 
 _use_enchant = Config.getOption("UseEnchant", None)
+_install_info_triggers = Config.getOption('InstallInfoTriggers', False)
+
 if _use_enchant or _use_enchant is None:
     try:
         import enchant
@@ -629,6 +631,9 @@ class TagsCheck(AbstractCheck.AbstractCh
 
             if d[1] == rpm.RPMSENSE_EQUAL and d[2][2] is not None:
                 printWarning(pkg, 'requires-on-release', value)
+            if _install_info_triggers:
+                if d[0] == 'info-install' or d[0].endswith('install-info'):
+                    printWarning(pkg, 'requires-on-install-info', value)
             self._unexpanded_macros(pkg, 'dependency %s' % (value,), value)
 
         self._unexpanded_macros(pkg, 'Name', name)
@@ -1115,6 +1120,10 @@ once.''',
 'tag-not-utf8',
 '''The character encoding of the value of this tag is not UTF-8.''',
 
+'requires-on-install-info',
+'''This rpm requires install-info which is no longer necessary as it's use now
+is automatically handled by a file trigger.''',
+
 'requires-on-release',
 '''This rpm requires a specific release of another package.''',
 
