--- rpmlint-1.4/config.info~	2011-06-10 17:42:09.000000000 +0200
+++ rpmlint-1.4/config	2012-07-28 00:28:41.326446258 +0200
@@ -206,7 +206,13 @@ from Config import *
 # Man page warning category, passed to groff -w while checking man pages.
 # See the groff(1) or troff(1) man pages for available categories.
 # Type: string, default: 'mac'
-#SetOption("ManWarningCategory", 'mac')
+#setOption("ManWarningCategory", 'mac')
+
+# Whether install-info of texinfo documentation is handled by triggers.
+# If handled by triggers, report if install-info is used in scriptlets and
+# also any dependencies on it.
+# Type: boolean, default: False
+#setOption("InstallInfoTriggers", False)
 
 # Output filters.
 # ---------------
--- rpmlint-1.4/FilesCheck.py.info~	2011-12-04 17:22:02.000000000 +0100
+++ rpmlint-1.4/FilesCheck.py	2012-07-28 00:29:24.425907442 +0200
@@ -220,6 +220,7 @@ skipdocs_regex = re.compile(Config.getOp
 meta_package_regex = re.compile(Config.getOption('MetaPackageRegexp', '^(bundle|task)-'))
 filesys_packages = ['filesystem'] # TODO: make configurable?
 quotes_regex = re.compile('[\'"]+')
+install_info_triggers = Config.getOption('InstallInfoTriggers', False)
 
 for idx in range(0, len(dangling_exceptions)):
     dangling_exceptions[idx][0] = re.compile(dangling_exceptions[idx][0])
@@ -592,21 +593,32 @@ class FilesCheck(AbstractCheck.AbstractC
 
                 # check install-info call in %post and %postun
                 if f.startswith('/usr/share/info/'):
-                    if not postin:
-                        printError(pkg,
-                                   'info-files-without-install-info-postin', f)
-                    elif not install_info_regex.search(postin):
-                        printError(pkg, 'postin-without-install-info', f)
+                    if not install_info_triggers:
+                        if not postin:
+                            printError(pkg,
+                                    'info-files-without-install-info-postin', f)
+                        elif not install_info_regex.search(postin):
+                            printError(pkg, 'postin-without-install-info', f)
+                    elif install_info_triggers:
+                        if postin and install_info_regex.search(postin):
+                            printError(pkg,
+                                    'info-files-with-install-info-postin', f)
 
                     preun = pkg[rpm.RPMTAG_PREUN] or \
                         pkg.scriptprog(rpm.RPMTAG_PREUNPROG)
-                    if not postun and not preun:
-                        printError(pkg,
-                                   'info-files-without-install-info-postun', f)
-                    elif (not postun or
-                          not install_info_regex.search(postun)) and \
-                          (not preun or not install_info_regex.search(preun)):
-                        printError(pkg, 'postin-without-install-info', f)
+                    if not install_info_triggers:
+                        if not postun and not preun:
+                            printError(pkg,
+                                    'info-files-without-install-info-postun', f)
+                        elif (not postun or
+                             not install_info_regex.search(postun)) and \
+                             (not preun or not install_info_regex.search(preun)): 
+                            printError(pkg, 'postin-without-install-info', f)
+                    elif install_info_triggers:
+                        if (postun and install_info_regex.search(postun)) or \
+                                (preun and install_info_regex.search(preun)):
+                            printError(pkg,
+                                    'info-files-with-install-info-postun', f)
 
                 # check perl temp file
                 if perl_temp_file_regex.search(f):
@@ -988,6 +1000,11 @@ a call to ldconfig.''',
 'postun-without-ldconfig',
 '''This package contains a library and its %postun doesn't call ldconfig.''',
 
+'info-files-with-install-info-postin',
+'''While this package contains info files and provides a %post scriptlet
+containing a call to install-info, this is no longer required due to file
+trigger replacement.''',
+
 'info-files-without-install-info-postin',
 '''This package contains info files and provides no %post scriptlet containing
 a call to install-info.''',
@@ -995,6 +1012,11 @@ a call to install-info.''',
 'postin-without-install-info',
 '''This package contains info files and its %post doesn't call install-info.''',
 
+'info-files-with-install-info-postun',
+'''While this package contains info files and provides a %postun scriptlet
+containing a call to install-info, this is no longer required due to file
+trigger replacement.''',
+
 'info-files-without-install-info-postun',
 '''This package contains info files and provides no %postun scriptlet containing
 a call to install-info.''',
--- rpmlint-1.4/TagsCheck.py.info~	2012-07-28 00:28:41.300446585 +0200
+++ rpmlint-1.4/TagsCheck.py	2012-07-28 00:28:41.329446222 +0200
@@ -26,6 +26,8 @@ import FilesCheck
 import Pkg
 
 _use_enchant = Config.getOption("UseEnchant", None)
+_install_info_triggers = Config.getOption('InstallInfoTriggers', False)
+
 if _use_enchant or _use_enchant is None:
     try:
         import enchant
@@ -617,6 +619,9 @@ class TagsCheck(AbstractCheck.AbstractCh
                     printError(pkg, 'explicit-lib-dependency', d[0])
             if d[1] == rpm.RPMSENSE_EQUAL and d[2][2] is not None:
                 printWarning(pkg, 'requires-on-release', value)
+            if _install_info_triggers:
+                if d[0] == 'info-install' or d[0].endswith('install-info'):
+                    printWarning(pkg, 'requires-on-install-info', value)
             self._unexpanded_macros(pkg, 'dependency %s' % (value,), value)
 
         self._unexpanded_macros(pkg, 'Name', name)
@@ -1094,6 +1099,10 @@ once.''',
 'tag-not-utf8',
 '''The character encoding of the value of this tag is not UTF-8.''',
 
+'requires-on-install-info',
+'''This rpm requires install-info which is no longer necessary as it's use now
+is automatically handled by a file trigger.''',
+
 'requires-on-release',
 '''This rpm requires a specific release of another package.''',
 
