--- rpmlint-1.7/Pkg.py.omv~	2016-08-19 12:29:18.688092037 +0200
+++ rpmlint-1.7/Pkg.py	2016-08-19 12:30:55.385639433 +0200
@@ -487,7 +487,7 @@ class Pkg(AbstractPkg):
         elif self.isSource():
             self.arch = 'src'
         else:
-            self.arch = self.header.format("%{ARCH}")
+            self.arch = self.header[rpm.RPMTAG_ARCH]
 
     # Return true if the package is a source package
     def isSource(self):
