--- rpmlint-1.7/TagsCheck.py.utf8changelog~	2016-08-19 12:11:34.354982584 +0200
+++ rpmlint-1.7/TagsCheck.py	2016-08-19 12:13:34.307415894 +0200
@@ -767,7 +767,7 @@ class TagsCheck(AbstractCheck.AbstractCh
                     changelog = changelog + clt
                 for s in changelog:
                     if not Pkg.is_utf8_bytestr(s):
-                        printError(pkg, 'tag-not-utf8', '%changelog')
+                        printWarning(pkg, 'tag-not-utf8', '%changelog')
                         break
 
             clt = pkg[rpm.RPMTAG_CHANGELOGTIME][0]
