--- SpecCheck.py.orig	2013-05-23 11:06:45.190475859 +0400
+++ SpecCheck.py	2013-05-23 11:09:55.668744646 +0400
@@ -17,7 +17,7 @@
 import rpm
 
 from Filter import addDetails, printError, printWarning
-from TagsCheck import VALID_GROUPS
+from TagsCheck import VALID_GROUPS, max_line_len, use_utf8
 import AbstractCheck
 import Config
 import Pkg
@@ -232,6 +232,14 @@
 
                 continue
 
+            if current_section == 'description' and not re.match(r'^\s*#', line):
+                utf8line = line[:-1]
+                if use_utf8:
+                    utf8line = Pkg.to_utf8(utf8line).decode('utf-8')
+                utf8line.strip
+                if len(utf8line) > max_line_len:
+                    printError(pkg,'description-line-too-long', line)
+
             if current_section in ('prep', 'build') and \
                     contains_buildroot(line):
                 printWarning(pkg, 'rpm-buildroot-usage', '%' + current_section,
