--- rpmlint-1.4/rpmlint.bero	2012-12-28 00:29:50.460489290 +0100
+++ rpmlint-1.4/rpmlint	2012-12-28 00:32:05.301834846 +0100
@@ -80,7 +80,14 @@ def loadCheck(name):
 #############################################################################
 def main():
 
-    locale.setlocale(locale.LC_COLLATE, '')
+    try:
+        locale.setlocale(locale.LC_COLLATE, '')
+    except locale.Error:
+        # This can happen if the locale files are corrupted, e.g. by
+        # having updated glibc, but not yet the locale-* packages.
+        # (Guaranteed to happen when building the locales for a new version
+        # of (e)glibc for the first time)
+        pass
 
     # Add check dirs to the front of load path
     sys.path[0:0] = Config.checkDirs()
