diff -up rpmlint-2.5.0/rpmlint/checks/PythonCheck.py.omv~ rpmlint-2.5.0/rpmlint/checks/PythonCheck.py
--- rpmlint-2.5.0/rpmlint/checks/PythonCheck.py.omv~	2023-11-26 05:25:08.983554789 +0100
+++ rpmlint-2.5.0/rpmlint/checks/PythonCheck.py	2023-11-26 05:26:43.604739094 +0100
@@ -201,6 +201,8 @@ class PythonCheck(AbstractFilesCheck):
 
         # Add pythonX-foo variants
         names += [f'python\\d*-{re.escape(i)}' for i in names]
+        # More importantly, virtual Provides: calling for the same
+        names += [f'python.*dist\({re.escape(i)}\)' for i in names]
         regex = '|'.join(names)
         # Support complex requirements like
         # (python310-jupyter-server >= 1.15 with python310-jupyter-server < 3)
