diff -Naur rpmlint-1.4/TagsCheck.py rpmlint-1.4.lic/TagsCheck.py
--- rpmlint-1.4/TagsCheck.py	2011-11-27 23:35:51.000000000 +0300
+++ rpmlint-1.4.lic/TagsCheck.py	2013-10-26 21:05:30.000000000 +0400
@@ -138,6 +138,44 @@
     'Non-distributable',
     'Proprietary',
     'Shareware',
+    
+    # More documentation and content licenses 
+    # and short names for some of already mentioned above
+    # (http://wiki.rosalab.ru/en/index.php/ROSA_License_List)
+    'CDL',
+    'CC-BY',
+    'CC-BY-SA',
+    'CC-BY-ND',
+    'FBSDDL',
+    'GFDL',
+    'IEEE',
+    'LDPL',
+    'OFSFDL',
+    'Open Publication',
+    'Public Use',
+    'Verbatim',
+    'CC0',
+    'DSL',
+    'DMTF',
+    'OAL',
+    'EFML',
+    'Free Art',
+    'GeoGratis',
+    'Green OpenMusic',
+    'BSD with advertising',
+    # Some more abbreviations mentioned in wiki
+    'ASL 1.0',
+    'ASL 1.1',
+    'ASL 2.0',
+    'MPLv1.1',
+    'MPLv1.0',
+    'MPLv2.0',
+    'GPL',
+    'LPPL',
+    'LPPL1.2',
+    'LPPL1.3',
+    'ZPL 2.1',
+    'BSD with Attribution',
     )
 
 BAD_WORDS = {
