--- BinariesCheck.py.orig	2012-10-22 13:03:30.848197183 +0400
+++ BinariesCheck.py	2012-10-22 13:04:41.601500542 +0400
@@ -409,7 +409,8 @@
                 res = numeric_dir_regex.search(f)
                 fn = res and res.group(1) or f
                 if f not in exec_files and not so_regex.search(f) and \
-                        not versioned_dir_regex.search(fn):
+                        not versioned_dir_regex.search(fn) and \
+                        (pkg.name.find("lib",0,0) == 0):
                     printError(pkg, 'non-versioned-file-in-library-package', f)
             if version and version != -1 and version not in pkg.name:
                 printError(pkg, 'incoherent-version-in-name', version)
