--- rpmlint-1.7/FilesCheck.py.slash~	2016-08-19 12:02:48.202403151 +0200
+++ rpmlint-1.7/FilesCheck.py	2016-08-19 12:06:21.897436713 +0200
@@ -230,6 +230,7 @@ filesys_packages = ['filesystem']  # TOD
 quotes_regex = re.compile('[\'"]+')
 install_info_triggers = Config.getOption('InstallInfoTriggers', False)
 legacy_mandriva_filetriggers = re.compile('^/var/lib/rpm/filetriggers/.*\.(filter|script)$')
+pkgconfig_regex = re.compile('^/usr/lib(64|x32)?/pkgconfig/.*\.pc$')
 start_certificate_regex = re.compile('^-----BEGIN CERTIFICATE-----$')
 start_private_key_regex = re.compile('^----BEGIN PRIVATE KEY-----$')
 
@@ -994,6 +995,16 @@ class FilesCheck(AbstractCheck.AbstractC
             if legacy_mandriva_filetriggers.search(f):
                 printWarning(pkg, 'legacy-mandriva-filetriggers', f)
 
+            if pkgconfig_regex.search(f):
+                chunk = None
+                istext = False
+                if os.access(pkgfile.path, os.R_OK):
+                    (chunk, istext) = peek(pkgfile.path, pkg, 2048)
+                    if istext:
+                        for line in chunk.splitlines():
+                            if line.find('//') >= 0 and not line.find('://') >= 0:
+                                printError(pkg, 'double-slash-in-path', f, line)
+
         if len(log_files) and not logrotate_file:
             printWarning(pkg, 'log-files-without-logrotate', sorted(log_files))
 
@@ -1374,6 +1385,12 @@ possibly the entire file with a new copy
 '''This package provides legacy Mandriva file triggers. These triggers should
 now be replaced by native rpm file triggers.''',
 
+'double-slash-in-path',
+'''This pkg-config file contains a path with double slash ('//') in it. This
+will break debugedit when stripping debug symbols during package building if
+these paths has been passed to gcc, and fail with the following error:
+"canonicalization unexpectedly shrank by one character"''',
+
 'gzipped-svg-icon',
 '''Not all desktop environments that support SVG icons support them gzipped
 (.svgz).  Install the icon as plain uncompressed SVG.''',
