--- rpmlint-1.4/Pkg.py.nosig~	2012-05-17 10:43:27.342078726 +0200
+++ rpmlint-1.4/Pkg.py	2012-05-17 10:43:30.635100274 +0200
@@ -435,7 +435,7 @@ class Pkg:
             # Create a package object from the file name
             ts = rpm.TransactionSet()
             # Don't check signatures here...
-            ts.setVSFlags(rpm._RPMVSF_NOSIGNATURES)
+            ts.setVSFlags(rpm.RPMVSF_NODSAHEADER | rpm.RPMVSF_NORSAHEADER |	rpm.RPMVSF_NODSA | rpm.RPMVSF_NORSA)
             fd = os.open(filename, os.O_RDONLY)
             try:
                 self.header = ts.hdrFromFdno(fd)
