--- rpmlint-1.5/BinariesCheck.py.usrlib_ldd~	2014-06-12 21:59:23.107661900 +0200
+++ rpmlint-1.5/BinariesCheck.py	2014-06-12 22:00:12.289047048 +0200
@@ -223,6 +223,14 @@ class BinaryInfo:
                         else:
                             in_unused = False
 
+        if not is_ar and not is_debug and (path.startswith('/bin') \
+                or path.startswith('/lib') or path.startswith('/sbin')):
+            res = Pkg.getstatusoutput(
+                ('env', 'LC_ALL=C', 'ldd', path))
+            for l in res[1].splitlines():
+                if '/usr' in l:
+                    printError(pkg, 'links-against-library-in-usr', file)
+
 path_regex = re.compile('(.*/)([^/]+)')
 numeric_dir_regex = re.compile('/usr(?:/share)/man/man./(.*)\.[0-9](?:\.gz|\.bz2)')
 versioned_dir_regex = re.compile('[^.][0-9]')
@@ -633,6 +641,11 @@ http://bugs.debian.org/cgi-bin/bugreport
 '''This executable must be position independent.  Check that it is built with
 -fPIE/-fpie in compiler flags and -pie in linker flags.''',
 
+'links-against-library-in-usr',
+'''Libraries and executables under /bin, /sbin, /lib and /lib64 may not link
+against libraries located under /usr as this might be a different partition
+not necessarily mounted at same time, ie. such as during boot.''',
+
 'missing-call-to-setgroups-before-setuid',
 '''This executable is calling setuid and setgid without setgroups or initgroups.
 There is a high probability this mean it didn't relinquish all groups, and this
