--- rpmlint-1.4/BinariesCheck.py.shlib_exec~	2012-03-02 19:36:30.862114477 +0100
+++ rpmlint-1.4/BinariesCheck.py	2012-03-02 19:36:33.406115745 +0100
@@ -330,6 +330,9 @@ class BinariesCheck(AbstractCheck.Abstra
                 for ec in bin_info.exit_calls:
                     printWarning(pkg, 'shared-lib-calls-exit', fname, ec)
 
+                if not (pkgfile.mode & stat.S_IEXEC):
+                    printWarning(pkg, 'shared-lib-not-executable', fname)
+
             # rpath ?
             if bin_info.rpath:
                 for p in bin_info.rpath:
@@ -545,6 +548,11 @@ state that the program has. It is prefer
 actual error code and let the calling program decide how to handle the
 situation.''',
 
+'shared-lib-not-executable',
+'''This library doesn't have the executable bit set. Without this bit set,
+rpm for instance won't be able identify the file as a library and not
+generate dependencies or strip debug symbols from it.''',
+
 'ocaml-mixed-executable',
 '''Executables built with ocamlc -custom are deprecated.  Packagers should ask
 upstream maintainers to build these executables without the -custom option.  If
