diff -Naur rpmlint-rpmlint-1.11/Pkg.py rpmlint-rpmlint-1.11.tpg/Pkg.py
--- rpmlint-rpmlint-1.11/Pkg.py	2020-08-20 11:01:34.804072896 +0000
+++ rpmlint-rpmlint-1.11.tpg/Pkg.py	2020-08-20 11:12:01.773076427 +0000
@@ -154,7 +154,7 @@
 
 bz2_regex = re.compile(r'\.t?bz2?$')
 xz_regex = re.compile(r'\.(t[xl]z|xz|lzma)$')
-
+zst_regex = re.compile(r'\.zst$')
 
 def catcmd(fname):
     """Get a 'cat' command that handles possibly compressed files."""
@@ -163,6 +163,8 @@
         cat = 'bzip2 -dcf'
     elif xz_regex.search(fname):
         cat = 'xz -dc'
+    elif zst_regex.search(fname):
+        cat = 'zstd -dc'
     return cat
 
 
diff -Naur rpmlint-rpmlint-1.11/SourceCheck.py rpmlint-rpmlint-1.11.tpg/SourceCheck.py
--- rpmlint-rpmlint-1.11/SourceCheck.py	2019-01-09 18:09:10.000000000 +0000
+++ rpmlint-rpmlint-1.11.tpg/SourceCheck.py	2020-08-20 11:09:51.043075691 +0000
@@ -25,6 +25,7 @@
     'gz': 'gzip compressed',
     'tgz': 'gzip compressed',
     'bz2': 'bzip2 compressed'
+    'zst': 'ZSTD compressed'
 }
 
 
